### DISCUSSION TITLE
Protest info sharing app

### DISCUSSION THREAD

**Discussion by farnoodma** [WeAreMahsaAmini/Core#23] [Ideas]
**Context**: In the middle of protesting, most probably the regime will cut off the internet in the area, and people do not have knowledge of the hot locations, where the police are, and where to go.

**State**: Right now, people are using social media to share this info, they also use the #الو hashtag of the Mamlekate channel to report such things to others.

**Idea**: If we can create a network of people who have the app installed and connect them with wifi, share the info p2p, and validate the location to allow people to report police traps, hot zone to join, etc. then without even an internet people can communicate in the middle of protesting.

**Comment by ghost** [comment]
The problem with p2p is the range, if groups of people are more than 100m apart, bluetooth won't work, so people in different regions can't communicate with each other. Alternative solutions:

1- Embed the time and location you want to share using zero-width characters (e.g. persian half space). Use one zero-width unicode character as 0, another as 1, and you can encode any binary data. Add this at the end of a generic comment (e.g. "great") and post it on some random blogfa weblog's comments section (or any other place you can publicly write text). I assume blogfa's servers are inside Iran, so it won't be affected by internet cut off. Then everyone else can crawl the comments of this weblog for location info.

2- Use an unpredictable but universally known information (e.g. stock price of an Iranian company at a specific time) as a random seed to generate a random protest path for the next hour. Since everyone uses the same seed, everyone gets the same path, and because the information is not known beforehand, the government can't prepare for this.

**Comment by theStratus** [comment]
it is a brilliant idea. I have seen a couple of apps out there for this. 
see this comment about [bridgefy.](https://bridgefy.me/) It is supposed to be used in Hong Kong protests as well. 
https://github.com/WeAreMahsaAmini/Core/discussions/66#discussioncomment-3748155
However, it is double-sided. Any real time info could be against people.

**Comment by nabardesarnevesht** [comment]
I have another solution for sharing routes to all people. 
https://github.com/nabardesarnevesht/NSI
This app receive push notifications but don't show them. Stores in its database and notify user at the specified time for each record.
User's segmented by province and city. So they receive targeted news.
You can [download ](https://nabardesarnevesht.github.io/) and test it
